src/Worker/Core/Shims/TaskOrchestrationContextWrapper.cs

```diff
@@ -482,4 +481,32 @@ async Task<T> InvokeWithCustomRetryHandler<T>(
             }
         }
     }
+
+    string GetDefaultVersion()
+    {
+        // The default version can come from two different places depending on the context of the invocation.
+        string? propertiesDefault;
+        string? optionsDefault;
+
+        if (this.invocationContext.Options.Versioning?.DefaultVersion is { } v)
+        {
+            optionsDefault = v;
```

Comment:
nit: if the options default is preferred, it would be more efficient to just return here rather than continue on to the next if check and pay the cost of a dictionary lookup.